### ISSUE TITLE
Replace integration in `set_cosmo_dist` with gsl integration

### ISSUE THREAD

**Issue by manodeep** [manodeep/Corrfunc#72]
Easy to replace. But would probably change the output of `DDrppi_mocks` and `vpf_mocks` and the `correct` result for the tests would have to be updated. 

Call to `set_cosmo_dist` should also then pass the max. `cz` required. Removes the need for macro `MAX_REDSHIFT_FOR_COSMO_DIST` set in `set_cosmo_dist.h`

**Comment by manodeep** [comment]
Since the mocks might be using arbitrary cosmologies and distributions, this change is essential.

**Comment by manodeep** [comment]
The tests for mocks (``DDrppi_mocks`` and ``vpf_mocks``) would probably need to be replaced - since the integrals will almost certainly not agree.

**Comment by manodeep** [comment]
The biggest issue with `set_cosmo_dist` was the fixed steps in `z`. I have now fixed that with 4afd728da160e3845f537b92616ec3d134495272. Maximum redshift is now passed in at runtime (and not fixed at compile time) -- see b8340295c27ef1a13934e2fc8905f8b3ceefb503. 

I also checked the integration in `set_cosmo_dist` with the gsl integration and the results are consistent to about 8-9 decimal places. 

No point any more for replacing `set_cosmo_dist` integration - closing for now.
